### ISSUE TITLE
Customer EAV Decimal Attribute required allow 0

### ISSUE THREAD

**Issue by nthurston** [magento/magento2#20030]
### Preconditions (*)

1. Magento Version 2.2.6

### Steps to reproduce (*)

1. Create a new required customer attribute of type decimal with a default of 0 e.g.
```
'bal' => [
  'is_used_in_grid' => false,
  'is_visible_in_grid' => false,
  'is_filterable_in_grid' => false,
  'is_searchable_in_grid' => false,
  'type' => 'decimal',
  'label' => 'AR Balance',
  'input' => 'text',
  'required' => true,
  'default' => 0,
  'sort_order' => 202,
  'visible' => true,
  'system' => false,
  'validate_rules' => null,
  'position' => 202,
  'user_defined' => false,
  'frontend_class' => 'validate-number'
],
```
2. Attempt to create a customer on front-end, or attempt to assign the value of 0 through the API.

### Expected result (*)

1. Customer attribute should be set with a 0 value.

### Actual result (*)

1. Both of these fail with [Attribute] is a required value.

**Comment by magento-engcom-team** [comment]
Hi @nthurston. Thank you for your report. 
To help us process this issue please make sure that you provided the following information:
- [ ] Summary of the issue
- [ ] Information on your environment
- [ ] Steps to reproduce
- [ ] Expected and actual results

Please make sure that the issue is reproducible on the vanilla Magento instance following **Steps to reproduce**. To deploy vanilla Magento instance on our environment, please, add a comment to the issue: 

`@magento-engcom-team give me $VERSION instance` 

where `$VERSION` is [version tags](https://github.com/magento/magento2/tags) (starting from 2.2.0+) or develop branches (for example: 2.3-develop). 
For more details, please, review the [Magento Contributor Assistant](https://devdocs.magento.com/guides/v2.2/contributor-guide/contributing.html#contributor-assist) documentation.

@nthurston do you confirm that you was able to reproduce the issue on vanilla Magento instance following steps to reproduce?
- [ ] yes
- [ ] no

**Comment by magento-engcom-team** [comment]
Hi @shikhamis11. Thank you for working on this issue. 
In order to make sure that issue has enough information and ready for development, please read and check the following instruction: :point_down:
- [ ] 1. Verify that issue has all the required information. (**Preconditions**, **Steps to reproduce**, **Expected result**, **Actual result**).<details><summary>Details</summary>If the issue has a valid description, the label `Issue: Format is valid` will be added to the issue automatically. Please, edit issue description if needed, until label `Issue: Format is valid` appears.</details>

- [ ] 2. Verify that issue has a meaningful description and provides enough information to reproduce the issue. If the report is valid, add `Issue: Clear Description` label to the issue by yourself.

- [ ] 3. Add `Component: XXXXX` label(s) to the ticket, indicating the components it may be related to.
- [ ] 4. Verify that the issue is reproducible on `2.3-develop` branch<details><summary>Details</summary>- Add the comment `@magento-engcom-team give me 2.3-develop instance` to deploy test instance on Magento infrastructure. <br/> - If the issue is reproducible on `2.3-develop` branch, please, add the label `Reproduced on 2.3.x`.<br/>- If the issue is not reproducible, add your comment that issue is not reproducible and close the issue and **_stop verification process here_**!
</details>

- [ ] 5. Verify that the issue is reproducible on `2.2-develop` branch. <details><summary>Details</summary>- Add the comment `@magento-engcom-team give me 2.2-develop instance` to deploy test instance on Magento infrastructure. <br/>- If the issue is reproducible on `2.2-develop` branch, please add the label `Reproduced on 2.2.x`</details>

- [ ] 6. Add label `Issue: Confirmed` once verification is complete. 

- [ ] 7. Make sure that automatic system confirms that report has been added to the backlog.

**Comment by shikhamis11** [comment]
@nthurston did you checked the same for catalog attributes?

**Comment by nthurston** [comment]
Just checked, this issue does not seem to be present for catalog attributes.  Looks to be isolated to customers.

Possibly related to #7753, but i would expect if I set a valid default value for a required attribute it should use the default.  E.g. I can create a required 'varchar' attribute with a default and not have it in the customer front-end registration form, which works fine.

**Comment by shikhamis11** [comment]
yes I am able to reproduce this issue

**Comment by magento-engcom-team** [comment]
@shikhamis11 Thank you for verifying the issue. Based on the provided information internal tickets `MAGETWO-97407` were created

**Comment by magento-engcom-team** [comment]
Hi @nthurston. Thank you for your report. 
The issue has been fixed in magento/magento2#20130 by @shikhamis11 in 2.3-develop branch
Related commit(s):
 - [53d861a9a4f429cb5971d65316dd00fa3a79ed7e](https://github.com/magento/magento2/commit/53d861a9a4f429cb5971d65316dd00fa3a79ed7e)
 - [1c464c2e7720dacf69b12249663a0b77603c18c5](https://github.com/magento/magento2/commit/1c464c2e7720dacf69b12249663a0b77603c18c5)
 - [44a21498df21c8112056da0d865251afbb7a521d](https://github.com/magento/magento2/commit/44a21498df21c8112056da0d865251afbb7a521d)

The fix will be available with the upcoming 2.3.1 release.
